backends/vulkan/runtime/graph/ops/impl/BinaryOp.cpp

```diff
@@ -75,14 +75,15 @@ void add_binary_op_node(
   const api::utils::ivec2 broadcast_params =
       create_broadcast_params(t_in1, t_in2);
 
-  std::stringstream kernel_name;
-  kernel_name << "binary_" << op_name;
-  apply_memory_layout_suffix(kernel_name, t_out);
-  apply_dtype_suffix(kernel_name, t_out);
+  std::string kernel_name("binary_");
+  kernel_name.reserve(kShaderNameReserve);
```

Comment:
nit: you would be slightly better off reserving before writing `"binary_"` into the string; as written, `"binary_"` might get written into the inline storage once before getting copied to the heap if the compiler doesn't catch it. surprisingly, this was very easy to demonstrate: clang 17 gets it wrong because reserve() isn't getting inlined. https://godbolt.org/z/jKcf3KGG1